2_process/src/combine_NLCD_PRMS.R

```diff
@@ -0,0 +1,58 @@
+###-----
+
+## Subset Land to AOI and join with PRMS ids and COMID area (sqkm)
+AOI_LC_w_area <- function(NLCD_LC_df, area_att, aoi_comids_df){
+  
+  #' @description following the comid id as the main key, this function subset NLCD LC data to the area of interest, joins with nhd COMID area info and PRMS info
+  #' Output is a df with PRMS ids, COMIDs, AREA(SQKM), and NLCD LC percentage per COMID (unique key)
+  #' @param area_att df with comids and associated area (AREASQKM)
+  #' @param NLCD_LC_df selected LC dataframe structured by COMID 
+  #' @param aoi_comids_df df linking comid and prms for the area of interest, already filter to area of interest 
+  #' @example AOI_LC_w_area(NLCD_LC_df = p1_NLCD_df$NLCD_LandCover_2011, area_att = p2_nhd_area_att, drb_comids_df = p2_drb_comids)
+  
+# Filter land cover attributes to aoi comids (drb) and join with area columns
+  lc_area <- NLCD_LC_df %>%
+    filter(COMID %in% aoi_comids_df$comid) %>%
+    left_join(., area_att, by="COMID") %>%
+    mutate(COMID = as.character(COMID))
+
+# Create df with PRMS, COMID, and NLCD LC att for the aoi (drb)
+  lc_aoi <- aoi_comids_df %>%
+    left_join(., lc_area, by = c('comid' = "COMID"))
+
+  return(lc_aoi)
+
+  }
+
+###----
+# Estimate prop of LC coverage at PRMS catchment ('CAT') scale
+proportion_lc_by_prms <- function(NLCD_LC_df_w_area){
+
+#' @description This function takes the consolidated df from AOI_LC_w_area() and calculated the proportion share of each LC class at the PRMS segid scale
+#' @param NLCD_LC_df_w_area df output of AOI_LC_w_area that has LC data per COMID & PRMS id and area in (km2)
+#' @example proportion_lc_by_prms(NLCD_LC_df_w_area = p2_LC_w_catchment_area)
+
+  area_df <- NLCD_LC_df_w_area %>%
+    mutate(AREASQKM_approx = case_when(AREASQKM == 0 ~ LENGTHKM^2, TRUE  ~ AREASQKM)) %>% 
+    # multiply COMID area by the LC percent for all the cols that start with 'CAT" - unit km2
+    mutate(across(starts_with('CAT'),~(.x/100)*AREASQKM_approx,.names="AREA_{col}")) %>%
+    # aggregate to PRMS scale through group_by()
+    group_by(PRMS_segid) %>%
+    # summarize group by at PRMS catchment scale
+    summarize(
+      # Sum to calculate total area of the PRMS  catchment - unit km2
+      across(AREASQKM_approx, sum, .names='AREASQKM_PRMS'),
+      # Sum to calculate  LC area at the PRMS catchment scale per LC class
+      across(starts_with(paste0('AREA_','CAT')), sum, .names = 'PRMS_{col}'),
+      # calculate proportion coverage of specified LC class in the entire PRMS catchment - proportion
+      across(starts_with(paste0('PRMS_AREA_','CAT')), ~(.x/AREASQKM_PRMS),
+             .names="PROP_{col}"),
+      # round the new cols 
+      across(starts_with(c('PROP','PRMS')), round, 2), 
+    .groups="drop")
```

Comment:
Do you think we'll want areas of each land use type? In your data cleaning step before, you had a step ` %>% select(-starts_with('PRMS_AREA_CAT'))` that is now omitted.

Reply:
Ohh right. Didn't remember i removed that. good find! ... Yes I removed that because I figured we'd keep just the proportion values and we weren't interested in the area values. I'll put back this col filter code statement because it makes the output smaller and significantly easier to work with.

Reply:
OK, sounds good to me since there are lots of columns with the land use data. I wasn't sure if you had decided otherwise, so just figured I'd double-check. Thanks for your reply : )